WalletWasabi/WabiSabi/Backend/Banning/Prison.cs

```diff
@@ -71,18 +68,18 @@ public void Punish(Inmate inmate)
 
 		public bool TryRelease(OutPoint utxo, [NotNullWhen(returnValue: true)] out Inmate? inmate)
 		{
-			lock (Lock)
+			if (Inmates.TryGetValue(utxo, out inmate))
```

Comment:
This should be very carefully explained why this is correct. I think it isn't.

Reply:
it is not correct.